contrib/guix/libexec/build.sh

```diff
@@ -240,7 +240,7 @@ mkdir -p "$OUTDIR"
 # CONFIGFLAGS
 CONFIGFLAGS="--enable-reduce-exports --disable-bench --disable-gui-tests --disable-fuzz-binary"
 case "$HOST" in
-    *linux*) CONFIGFLAGS+=" --enable-glibc-back-compat" ;;
+    *linux*) CONFIGFLAGS+=" --disable-threadlocal" ;;
```

Comment:
Doesn't `configure` check `thread_local` support?

Maybe
```diff
diff --git a/configure.ac b/configure.ac
index 94f01a51a..ee8d716e5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1027,10 +1027,7 @@ AC_COMPILE_IFELSE([AC_LANG_SOURCE([
   [AC_MSG_RESULT(no)]
 )
 
-dnl thread_local is currently disabled when building with glibc back compat.
-dnl Our minimum supported glibc is 2.17, however support for thread_local
-dnl did not arrive in glibc until 2.18.
-if test "x$use_thread_local" = xyes || { test "x$use_thread_local" = xauto && test "x$use_glibc_compat" = xno; }; then
+if test "x$use_thread_local" != xno; then
   TEMP_LDFLAGS="$LDFLAGS"
   LDFLAGS="$TEMP_LDFLAGS $PTHREAD_CFLAGS"
   AC_MSG_CHECKING([for thread_local support])
diff --git a/contrib/guix/libexec/build.sh b/contrib/guix/libexec/build.sh
index 02ef80096..fd76cb2b7 100755
--- a/contrib/guix/libexec/build.sh
+++ b/contrib/guix/libexec/build.sh
@@ -239,9 +239,6 @@ mkdir -p "$OUTDIR"
 
 # CONFIGFLAGS
 CONFIGFLAGS="--enable-reduce-exports --disable-bench --disable-gui-tests --disable-fuzz-binary"
-case "$HOST" in
-    *linux*) CONFIGFLAGS+=" --disable-threadlocal" ;;
-esac
 
 # CFLAGS
 HOST_CFLAGS="-O2 -g"
```
?

Reply:
The change you've suggested is not going to work for supporting glibc 2.17 at runtime, for the exact reason outlined in the comment you've deleted.

Reply:
The `configure` script in Guix environment with glibc 2.17 should leave `HAVE_THREAD_LOCAL` undefined after the failed check, no?

Reply:
> Guix environment with glibc 2.17

The Guix environment is based on [glibc 2.24](https://github.com/bitcoin/bitcoin/blob/692b78793bda5a06bd0f7907d6f62286f53c8e74/contrib/guix/manifest.scm#L574), except for RISCV (2.27).

Reply:
Indeed.